client/App.js

```diff
@@ -8,29 +8,143 @@ class App extends Component {
     this.state = {
       courseId: 1,
       reviews: [],
+      allReviews: [],
+      showNumber: 5,
+      search: '',
     }
   }
+  
+  handdleChange(e){
+    e.preventDefault()
+    this.setState({
+      search: e.target.value
+    })
+    var courseId = this.state.courseId;
```

Comment:
I think it's best practices to adhere to a declaration style, var or let, or using them according to relevant situations. Otherwise it could confuse whoever maintains the code afterwards